### PR TITLE
Azure secret scoping

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
Add secret scoping feature to the Azure Secrets Manager.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/mlops-stacks/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/azure/secrets_managers/azure_secrets_manager.py [hunk 1]
```diff
@@ -139,59 +296,82 @@ def update_secret(self, secret: BaseSecretSchema) -> None:
 
         Args:
             secret: the secret to update
+
+        Raises:
+            KeyError: if the secret does not exist
         """
+        self.validate_secret_name(secret.name)
+        self._create_or_update_secret(secret)
```
Comment (schustmi) [inline src/zenml/integrations/azure/secrets_managers/azure_secrets_manager.py:304]: Unless my GitHub UI is bugging here, this call is duplicated from below and should probably not be called ahead of the following checks?

**Review by schustmi — APPROVED** [review]
LGTM!

File: src/zenml/integrations/azure/secrets_managers/azure_secrets_manager.py [hunk 2]
```diff
@@ -55,6 +65,109 @@ def _ensure_client_connected(cls, vault_name: str) -> None:
...
+                self.CLIENT.update_secret_properties(
+                    azure_secret_name,
+                    tags={
+                        ZENML_GROUP_KEY: secret.name,
+                        ZENML_KEY_NAME: key,
+                        ZENML_SCHEMA_NAME: secret.TYPE,
+                    },
+                )
+
+                logger.debug("Wrote secret: %s", azure_secret_name)
```
Comment (AlexejPenner) [inline src/zenml/integrations/azure/secrets_managers/azure_secrets_manager.py:156]:
```suggestion
                logger.debug("Secret `%s` written to the Azure Key Vault.", azure_secret_name)
```

Log output is a bit terse

**Review by AlexejPenner — APPROVED** [review]
LGTM! - one small CR for a debug log

**Pull request merged**